flang/lib/Lower/ConvertVariable.cpp

```diff
@@ -1646,8 +1646,8 @@ void Fortran::lower::genDeclareSymbol(
     fir::FirOpBuilder &builder = converter.getFirOpBuilder();
     const mlir::Location loc = genLocation(converter, sym);
     fir::FortranVariableFlagsAttr attributes =
-        Fortran::lower::translateSymbolAttributes(builder.getContext(), sym,
```

Comment:
This will lead to situations where the VOLATILE/ASYNCHRONOUS attributes are nor propagated to the hlfir.declare (these attributes can be added when using module variables). More complex logic inside translateSymbolAttributes is likely needed. We do not support these attributes yet, so fixing the OpenMP case is more important, but  could you add a FIXME about VOLATILE/ASYNCHRONOUS above?

Reply:
Done.